### ISSUE TITLE
Merge warnings reported for JSONObject custom scalar

### ISSUE THREAD

**Issue by vigie** [apollographql/apollo-client#7071]
**Intended outcome:**
My understanding is that all scalar values (custom or otherwise) should be treated opaquely and replaced when updating. Such updates should not trigger a data loss merge warning.

**Actual outcome:**
When merging objects containing fields of custom scalar type JSONObject, merge warning are generated in the console for these fields.

**How to reproduce the issue:**
Include [JSONObject custom scalar](https://github.com/Urigo/graphql-scalars) into your schema and resolver map and use it in a type definition

```
scalar JSONObject

type Deployment {
   id: ID!
   state: JSONObject
}
```
execute a query to fetch an object that uses JSONObject, then update the data in the cache so the merge algorithm runs.

```
query DeploymentQuery($id: ID) {
    deployment(id: $id)  {
        id
        state
    }
}
```

Notice that when a deployment object that has a different value for state is being merged, you see warnings in the console about potential data loss for `state` field.

**Versions**
```
  System:
    OS: macOS 10.15.6
  Binaries:
    Node: 12.18.2 - ~/.nvm/versions/node/v12.18.2/bin/node
    npm: 6.14.5 - ~/.nvm/versions/node/v12.18.2/bin/npm
  Browsers:
    Chrome: 85.0.4183.121
    Firefox: 79.0
    Safari: 14.0
  npmPackages:
    @apollo/client: ^3.2.0 => 3.2.0
    apollo-angular: ^2.0.3 => 2.0.4
```

**Comment by benjamn** [comment]
@vigie This should be fixed in `@apollo/client@3.3.0-beta.5` (which includes #7075).

To be clear, this means you shouldn't need to configure `JSONObject` scalar fields with `merge: false` to prevent warnings, because there shouldn't be any warnings when updating scalar fields.

**Comment by vigie** [comment]
Thanks for the quick response @benjamn, I can confirm the issue appears to be fixed in `3.3.0-beta.5` 👍
